runtime/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaPublishMetadata.java

```diff
@@ -93,20 +93,35 @@ public static final class KafkaOffsetMetadata
         public final IntArrayList packetIds;
 
         public long sequence;
+        public String topic;
 
         KafkaOffsetMetadata(
+            String topic,
             long producerId,
             short producerEpoch)
         {
-            this(producerId, producerEpoch, new IntArrayList());
+            this(topic, producerId, producerEpoch, new IntArrayList());
+        }
+
+        KafkaOffsetMetadata(
+            long producerId,
+            short producerEpoch,
+            IntArrayList packetIds)
+        {
+            this.sequence = 1;
+            this.producerId = producerId;
+            this.producerEpoch = producerEpoch;
+            this.packetIds = packetIds;
         }
 
         KafkaOffsetMetadata(
+            String topic,
             long producerId,
             short producerEpoch,
             IntArrayList packetIds)
         {
             this.sequence = 1;
+            this.topic = topic;
```

Comment:
I think we'll have the same issue unless this is changed, agree?
```suggestion
            this.topic = topic.intern();
```